quartz-plugins/src/main/java/org/quartz/plugins/interrupt/JobInterruptMonitorPlugin.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2001-2009 Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.quartz.plugins.interrupt;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.quartz.JobExecutionContext;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.Trigger.CompletedExecutionInstruction;
+import org.quartz.listeners.TriggerListenerSupport;
+import org.quartz.spi.ClassLoadHelper;
+import org.quartz.spi.SchedulerPlugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This plugin catches the event of job running for a long time (more than the
+ * configured max time) and tells the scheduler to "try" interrupting it if
+ * enabled.
+ * 
+ * @see org.quartz.Scheduler#interrupt()
+ * 
+ * @author Rama Chavali
+ */
+
+public class JobInterruptMonitorPlugin extends TriggerListenerSupport implements SchedulerPlugin {
+
+    private static final String JOB_INTERRUPT_MONITER_KEY = "JOB_INTERRUPT_MONITER_KEY";
+
+    private long DEFAULT_MAX_RUNTIME = 300000;
+
+    private ScheduledExecutorService executor;
+
+    @SuppressWarnings("rawtypes")
+    private ScheduledFuture future;
+
+    private Scheduler scheduler;
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    public JobInterruptMonitorPlugin() {
+    }
+
+    @Override
+    public void start() {
+    }
+
+    @Override
+
+    public void shutdown() {
+        this.executor.shutdown();
+    }
+
+    protected Logger getLog() {
+        return log;
+    }
+
+    @SuppressWarnings("rawtypes")
+    public ScheduledFuture scheduleJobInterruptMonitor(JobKey jobkey, long delay) {
+        return this.executor.schedule(new InterruptMonitor(jobkey, scheduler), delay, TimeUnit.MILLISECONDS);
+    }
+
+    // Bean Property Methods
+
+    public long getDefaultMaxRunTime() {
+        return this.DEFAULT_MAX_RUNTIME;
+    }
+
+    public void setDefaultMaxRunTime(long defaultMaxRunTime) {
+        this.DEFAULT_MAX_RUNTIME = defaultMaxRunTime;
+    }
+
+    // Trigger Listener Methods
+    public String getName() {
+        return "JobInterruptMonitorListner";
```

Comment:
Name shouldn't be hard-coded, should be obtained/remembered from the value passed in on the initialize method.

(also there's a typo in the spelling of this hard-coded name)